semgrep-core/Parsing/other/yaml_to_generic.ml

```diff
@@ -12,11 +12,6 @@ exception ParseError of string
  * low-level Stream API, which we parse into a generic AST
 *)
 
-type env = {
```

Comment:
I think you should keep this type, because it would be easier to extend.
I just didn't want to expose this internal type in the .mli. This is internal details that users
of the module should not care about.

Reply:
If I don't put it in the mli, how do I create the env to pass in when I call the function in parse_target?

Reply:
This is what I had in mind: https://github.com/returntocorp/semgrep/pull/2670/files